internal/command/deploy/deployv2.go

```diff
@@ -0,0 +1,388 @@
+
+	if appConfig.PrimaryRegion != "" && appConfig.Env["PRIMARY_REGION"] == "" {
+		appConfig.Env["PRIMARY_REGION"] = appConfig.PrimaryRegion
+	}
+
+	if deployToMachines {
+		ctx, err = command.LoadAppV2ConfigIfPresent(ctx)
+		if err != nil {
+			return fmt.Errorf("error loading appv2 config: %w", err)
+		}
+		primaryRegion := appConfig.PrimaryRegion
+		if flag.GetString(ctx, flag.RegionName) != "" {
+			primaryRegion = flag.GetString(ctx, flag.RegionName)
+		}
+		md, err := NewMachineDeployment(ctx, MachineDeploymentArgs{
+			AppCompact:        appCompact,
+			DeploymentImage:   img,
+			Strategy:          flag.GetString(ctx, "strategy"),
+			EnvFromFlags:      flag.GetStringSlice(ctx, "env"),
+			PrimaryRegionFlag: primaryRegion,
+			BuildOnly:         flag.GetBuildOnly(ctx),
+			SkipHealthChecks:  flag.GetDetach(ctx),
+			WaitTimeout:       time.Duration(flag.GetInt(ctx, "wait-timeout")) * time.Second,
+			LeaseTimeout:      time.Duration(flag.GetInt(ctx, "lease-timeout")) * time.Second,
+		})
+		if err != nil {
+			return err
+		}
+		return md.DeployMachinesApp(ctx)
+	}
+
+	release, releaseCommand, err = createReleaseV2(ctx, appConfig, img)
+	if err != nil {
+		return err
+	}
+
+	if flag.GetDetach(ctx) {
+		return nil
+	}
+
+	// TODO: This is a single message that doesn't belong to any block output, so we should have helpers to allow that
+	tb := render.NewTextBlock(ctx)
+	tb.Done("You can detach the terminal anytime without stopping the deployment")
+
+	// Run the pre-deployment release command if it's set
+	if releaseCommand != nil {
+		// TODO: don't use text block here
+		tb := render.NewTextBlock(ctx, fmt.Sprintf("Release command detected: %s\n", releaseCommand.Command))
+		tb.Done("This release will not be available until the release command succeeds.")
+
+		if err := watch.ReleaseCommand(ctx, appConfig.AppName, releaseCommand.ID); err != nil {
+			return err
+		}
+
+		release, err = apiClient.GetAppReleaseNomad(ctx, appConfig.AppName, release.ID)
+		if err != nil {
+			return err
+		}
+	}
+
+	if release.DeploymentStrategy == "IMMEDIATE" {
+		logger := logger.FromContext(ctx)
+		logger.Debug("immediate deployment strategy, nothing to monitor")
+
+		return nil
+	}
+
+	err = watch.Deployment(ctx, appConfig.AppName, release.EvaluationID)
+
+	return err
+}
+
+func useMachinesConfigV2(appConfig *appv2.Config, appCompact *api.AppCompact, args DeployWithConfigArgs) (bool, error) {
+	switch {
+	case appCompact.PlatformVersion == appv2.AppsV2Platform:
+		return true, nil
+	case appCompact.Deployed:
+		return appCompact.PlatformVersion == appv2.AppsV2Platform, nil
+	case args.ForceNomad:
+		return false, nil
+	case args.ForceMachines:
+		return true, nil
+	case len(appConfig.Statics) > 0:
+		// statics are not supported in Apps v2 yet
+		return false, nil
+	case args.ForceYes:
+		// if running automated, stay on nomad platform for now
+		return false, nil
+	default:
+		// choose nomad for now if not otherwise specified
+		return false, nil
+	}
+}
```

Comment:
Just a note that this will probably create a merge conflict with #1802. not a problem, just making a paper trail so it's easier to track down later :)

Reply:
Thank you :)

Reply:
Now that #1802 has merged, this'll have to be updated. Here's what the new version looks like :)
https://github.com/superfly/flyctl/blob/ad1f3de704c1a01ee588b0ca9839987953d37a1d/internal/command/deploy/deploy.go#L228-L244